src/packs/nzism-v36-1022-20.ts

```diff
@@ -0,0 +1,845 @@
+  S3BucketLoggingEnabled,
+  S3BucketPublicReadProhibited,
+  S3BucketPublicWriteProhibited,
+  S3BucketServerSideEncryptionEnabled,
+  S3BucketSSLRequestsOnly,
+  S3BucketVersioningEnabled,
+  S3DefaultEncryptionKMS,
+} from '../rules/s3';
+import {
+  SageMakerEndpointConfigurationKMSKeyConfigured,
+  SageMakerNotebookInstanceKMSKeyConfigured,
+  SageMakerNotebookNoDirectInternetAccess,
+} from '../rules/sagemaker';
+import { SecretsManagerUsingKMSKey } from '../rules/secretsmanager';
+import { SNSEncryptedKMS } from '../rules/sns';
+import {
+  VPCFlowLogsEnabled,
+  VPCDefaultSecurityGroupClosed,
+} from '../rules/vpc';
+import { WAFv2LoggingEnabled } from '../rules/waf';
+
+/**
+ * Check for NZISM v36-1022-20 compliance.
+ * Based on the NZISM v36-1022-20: <URL To be updated once published>
+ *
+ * The following rules are not yet included and are TODO items.
+ *
+ * CloudfrontDefaultRootObjectConfigured
+ * DynamodbTableEncryptedKms
+ * EbsSnapshotPublicRestorableCheck
+ * Ec2HardeningAndStateManagement
+ * Ec2Imdsv2Check
+ * EcsContainersReadonlyAccess
+ * ElbCustomSecurityPolicySslCheck
+ * EmrMasterNoPublicIp
+ * RdsSnapshotEncrypted
+ * VpcSgOpenOnlyToAuthorizedPorts
+ *
+ * A number of other checks from the NZISM Conformance Packs are account level checkes, and are excluded from stack level checks.
+ *
+ *
+ * */
+
+export class NZISM36Checks extends NagPack {
+  constructor(props?: NagPackProps) {
+    super(props);
+    this.packName = 'NZISM.36.1022.20';
+  }
+  public visit(node: IConstruct): void {
+    if (node instanceof CfnResource) {
+      this.checkELB(node);
+      this.checkAPIGW(node);
+      this.checkCloudTrail(node);
+      this.checkCloudFront(node);
+      this.checkKMS(node);
+      this.checkCloudWatch(node);
+      this.checkRDS(node);
+      this.checkECS(node);
+      this.checkEFS(node);
+      this.checkDMS(node);
+      this.checkDynamoDB(node);
+      this.checkEC2(node);
+      this.checkElasticBeanstalk(node);
+      this.checkElastiCache(node);
+      this.checkOpenSearch(node);
+      this.checkIAM(node);
+      this.checkLambda(node);
+      this.checkRedshift(node);
+      this.checkS3(node);
+      this.checkSageMaker(node);
+      this.checkSecretsManager(node);
+      this.checkSNS(node);
+      this.checkVPC(node);
+      this.checkWAF(node);
+    }
+  }
+
+  /**
+   * Check API Gateway Resources
+   * @param node the CfnResource to check
+   * @param ignores list of ignores for the resource
+   */
+  private checkAPIGW(node: CfnResource): void {
+    this.applyRule({
+      info: 'SHOULD 16.6.10.C.02[CID:2013], MUST 23.5.11.C.01[CID:7496]| Access Control and Passwords/Event Logging and Auditing/Additional events to be logged & Public Cloud Security/Logging and Alerting in Public Cloud/Logging requirements.',
+      explanation:
+        'API Gateway logging displays detailed views of users who accessed the API and the way they accessed the API. This insight enables visibility of user activities.',
```

Comment:
The `info` field should [match the format of other compliance packs](https://github.com/cdklabs/cdk-nag/blob/main/src/packs/nist-800-53-r5.ts#L183-L190). 

It's important for the `info` field to describe what triggered the rule ex. `The API Gateway stage does not have execution logging enabled for all methods`. The other compliance packs list the relevant control IDs at the end of the `info` field and we should do something *similar* here.

I don't think we need to describe what each control ID means here (`Access Control and Passwords/Event Logging and Auditing/Additional events to be logged & Public Cloud Security/Logging and Alerting in Public Cloud/Logging requirements`) since it will make the output very hard to read. I think if we list the controls users can lok up the reference.

Reply:
Agreed.   FYI, the format of the Controls  

SHOULD 16.6.6.C.02[CID:1998]

SHOULD is the directive,   the first number is the reference to the section, paragraph, subsection in the NZISM documentation, and  the CID is the NZISM Control ID.       

Will modify them to look like this. 

`this.applyRule({
      info: 'The trail does not have encryption enabled | SHOULD 17.1.46.C.04[CID:2082]',
      explanation:
        'Because sensitive data may exist and to help protect data at rest, ensure encryption is enabled for your AWS CloudTrail trails.',
      level: NagMessageLevel.WARN,
      rule: CloudTrailEncryptionEnabled,
      node: node,
    });`

Reply:
When taking a look at the [Config Conformance Pack for the previous revision](https://docs.aws.amazon.com/config/latest/developerguide/operational-best-practices-for-nzism.html), only the NZISM Control IDs are listed.

It also looks like the CIDs are unique within the [NZISM Information Security Manual](https://www.nzism.gcsb.govt.nz/assets/NZISM/pdf/NZISM-V.3.6-September-2022.pdf)

For the sake of both uniformity with other packs and verbosity I propose we do something like

`info: blah blah blah - (Control IDs: SHOULD[7777, 8888], MUST[9999]).`

or just

`info: blah blah blah - (Control IDs: 7777, 8888,  9999).`

I prefer the second approach since the `Warning` and `Error` level  provides a direct mapping to the highest directive specified by the CIDs, but I can be convinced of the former

Reply:
I think having both the document refernece and and the actual control ID is very helpful..  And the directive.    It will save me a lot of time, flipping between them.  The conformance Pack ( which does not 1:1 map to rules ) contains all this information..    For the sake of a few extra characters in the info,  its seems like a very small price to pay for a lot of useful information )..

(https://github.com/awslabs/aws-config-rules/blob/master/aws-config-conformance-packs/Operational-Best-Practices-for-NZISM.yaml)